docs/verifier/query-builder.md

```diff
@@ -2,59 +2,209 @@
+
+It's important to consider the proof type as it directly influences the circuit options provided, ensuring alignment with the verification method (off-chain or on-chain via a smart contract) and the specific requirements of the query.
+
+:::caution
+V3 circuits are experimental as of this writing, so users should exercise caution and stay informed about their development and stability.
+:::
+
+### Proof of Uniqueness
+
+The Proof of Uniqueness enhances security and integrity in contexts requiring [sybil resistance](https://en.wikipedia.org/wiki/Sybil_attack). Using this feature is optional when building a query and it's exclusively supported by v3 circuits. It's managed through two fields in the Query Builder form:
+
+1. **Check Proof of Uniqueness:** This checkbox is initially unchecked. When selected, it enables the Proof of Uniqueness functionality.
+
+2. **Nullifier Session ID:** A numeric input field that remains disabled unless the Proof of Uniqueness checkbox is checked. It requires the user to provide a nullifier session ID.
+
+The nullifier session ID field is a critical component for calculating the SybilID, which is essential for establishing Proof of Uniqueness in a query. This ID serves as a verifier-specific session identifier, which might represent, for example, a voting ID. It is used in the SybilID calculation formula:
+
+`SybilID = hash(genesisID, credProfileNonce, schemaHash, verifierID, nullifierSessionID)`
+
+Providing the correct nullifier session ID ensures the query's integrity and uniqueness, conforming to the protocol's sybil resistant features:
+
+- **One Identity Per Session:** Each identity can obtain only one identifier per credential and verifier session ID (CRS), consistent across all Profiles and Genesis ID.
+- **Brute-force Resistance:** Prevents brute-forcing to reveal Genesis ID or other values.
+- **Tracking Resistance:** Ensures Profile IDs and Genesis ID cannot be linked based on the Sybil ID.
+- **Flexible Session Duration:** Allows the verifier to keep the session active as needed, without time-specific constraints.
+- **Protection Against Collusion:** Prevents matching different identifiers as belonging to the same identity, even if the issuer and verifier collude.
+
+### Query Type
+
+The protocol allows selection between two query types:
+
+1. **Condition:** This type allows the prover to demonstrate that a specific condition is met without revealing the underlying data. It maintains data privacy by cryptographically verifying the condition. An example is proving legal age for entry into a venue without disclosing the actual age.
+
+2. **Selective disclosure:** Use this when the prover needs to reveal specific data stored in their credential. This option discloses the actual value of an attribute to the verifier. For instance, instead of just proving legal age, it would show the verifier the prover's actual age.
```

Comment:
> 2. **Selective disclosure:** Use this when you need the prover to reveal specific data stored in user credential. This option discloses the actual value of an attribute to the verifier. For instance, instead of just requesting a proof proving legal age, it would disclose user's actual age to the verifier.

Reply:
The `prover` and the `user` are the same here. I'd not use two terms.

> 2. **Selective disclosure:** Use this when you need the prover to reveal specific data stored in their credential. This option discloses the actual value of an attribute to the verifier. For instance, instead of just requesting proof proving legal age, it would disclose the prover's actual age to the verifier.

Reply:
Done